ludwig/combiners/combiners.py

```diff
@@ -106,11 +149,24 @@ def __init__(
         if input_features and len(input_features) == 1 and fc_layers is None:
             self.supports_masking = True
 
+    @property
+    def _input_last_dimension_size(self) -> int:
+        # compute the size of the last dimension for the incoming encoder outputs
+        # this is required to setup the fully connected layer
+        if self.flatten_inputs:
+            shapes = [np.prod(self.input_features[k].output_shape) for k in
+                      self.input_features]
```

Comment:
I believe these output shapes are all torch.Size, do we need numpy for doing this multiplication?

Reply:
If the thought is to use `torch.prod()` instead of `np.prod()`, I tried it first and received this error
```
>   shapes = [torch.prod(self.input_features[k].output_shape) for k in
              self.input_features]
E   TypeError: prod(): argument 'input' (position 1) must be Tensor, not torch.Size
```
This is how I ended up using numpy.   Let me know if I'm missing something.

Reply:
@w4nderlust 
As I understand the guidance, the recommendation is to use the `torch.numel()` method.  To get this to work, I have to create a short-lived `Tensor` with the required shape, like this:
```
hapes = [torch.numel(torch.zeros(self.input_features[k].output_shape)) for k in
                  self.input_features]
        return torch.Size([sum(shapes)])
```

I can't use `torch.numel()` directly on the `output_shape` property because I get this error:
```
>   shapes = [torch.numel(self.input_features[k].output_shape) for k in
              self.input_features]
E   TypeError: numel(): argument 'input' (position 1) must be Tensor, not torch.Size
```

To make use of `torch.numel()` I have to replace the one call to `np.prod()` with this nested call that creates a short-lived tensor of zeros: `torch.numel(torch.zeros(.....))`.  Right now it is not clear what the performance overhead is for creating the short-lived zero tensors vs using the NumPy function.

I'll make the requested change to `torch.numel()`.  I just want to be explicit on what I think may be performance considerations when using this with large encoder outputs.

Do you still want me to convert to `torch.numel()`?

Reply:
I may have misread th thread. i thought the `shape` object they referred to was actually a `torch.Size` so that you could call `torch.Size.numel()`.
If that's not feasible, `torch.prod(torch.Tensor(self.input_features[k].output_shape))` would do, but I agree with you, it would be nice to have a function that works on `Size` without the need to instantiate a new `Tensor`. (I believe internally `np.prod` very likely instantiates a `ndarray` anyway).

Reply:
converted to use of `torch.prod(torch.Tensor(....))` approach.  Removed use of `NumPy`.